File: church.rb

if_lambda = -> (b) { -> (x) { -> (y) { b.call(x).call(y) } } }

true_lambda  = -> (x) { -> (y) { x } }
false_lambda = -> (x) { -> (y) { y } }

puts "if True  a b = #{if_lambda.call(true_lambda).call("a").call("b")}"  # a
puts "if False a b = #{if_lambda.call(false_lambda).call("a").call("b")}" # b

File: church.js
let if_lambda = b => {
  return x => {
    return y => {
      return b(x)(y);
    };
  };
}

let true_lambda  = x => {
  return y => {
    return x;
  };
};

let false_lambda = x => {
  return  y => {
    return y;
  };
};

console.log("if True  a b =", if_lambda(true_lambda)("a")("b"));
console.log("if False a b =", if_lambda(false_lambda)("a")("b"));
